lnwire/extra_bytes.go

```diff
@@ -27,7 +33,13 @@ func (e *ExtraOpaqueData) Encode(w io.Writer) error {
 
 // Decode attempts to unpack the raw bytes encoded in the passed io.Reader as a
 // set of extra opaque data.
-func (e *ExtraOpaqueData) Decode(r io.Reader) error {
+func (e *ExtraOpaqueData) Decode(r io.Reader, pver uint32) error {
+	// Only if we're using the modern protocl version will we attempt to
+	// keep on decoding past the end of the "main message".
+	if pver != ProtocolVersionTLV {
+		return nil
```

Comment:
also, doesn't this change the behavior for the legacy protocol? Before this PR the messages that had `ExtraOpaqueData` defined would be read to the end, while after this we will stop reading (so if migrations are done for these messages, that data won't be copied).